meta-balena-common/classes/kernel-balena.bbclass

```diff
@@ -947,8 +956,54 @@ do_kernel_resin_checkconfig[dirs] += "${WORKDIR} ${B}"
 # Force compile to depend on the last resin task in the chain
 do_compile[deptask] += "do_kernel_resin_checkconfig"
 
-# copy to deploy dir latest .config and Module.symvers (after kernel modules have been built)
+do_configure:append () {
+    mkdir -p certs
+    if [ "${SIGN}" != "true" ]; then
+        return 0
+    fi
+
+    export CURL_CA_BUNDLE="${STAGING_DIR_NATIVE}/etc/ssl/certs/ca-certificates.crt"
+
+    RESPONSE_FILE=$(mktemp)
+    curl --fail "${SIGN_API}/kmod/cert/${SIGN_KMOD_KEY_ID}" > "${RESPONSE_FILE}"
+    jq -r .cert "${RESPONSE_FILE}" > certs/balenaos.crt
+    rm -f "${RESPONSE_FILE}"
```

Comment:
This is `kmod.cert` that `balena-keys` stores in the deploy folder.

Reply:
@mtoman this `balenaos.cert` appears unused, where is it used?

Reply:
I am thinking that whatever uses this `balenaos.cert` should directly use `$DEPLOYDIR/kmod.cert` instead.

Reply:
The kernel config refers to it which embeds it in the kernel. This is in the device PR at this moment: https://github.com/balena-os/balena-intel/blob/0de9a811a36628b5d35aa55610dd3ea128e53133/layers/meta-balena-genericx86/recipes-kernel/linux/linux-yocto_%25.bbappend#L434

The pull can be safely removed from here, now that we have balena-keys we can just pull it from there.

Reply:
And to be clear it is the kmod cert

Reply:
thanks @mtoman I'll remove it from here then. The device tree kernel PR will need to add `balena-keys` as a dependency and use it from `${DEPLOYDIR}/kmod.cert`